File: Beecrowd/2-AD-HOC/1250-KiloMan-2/1250.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int repeat;scanf("%d", &repeat);
    int ans, size;
    string jumps;
    for(int r =0;r<repeat;r++){
        ans=0;
        scanf("%d", &size);
        vector<int> disparos(size);
        for(int i =0;i<size;i++)scanf("%d", &disparos[i]);
        cin>>jumps;
        for(int i =0;i<size;i++){
            if(jumps[i] == 'S' && disparos[i]<=2)ans++;
            else if(jumps[i] == 'J' && disparos[i]>2)ans++;
        }
        printf("%d\n", ans);
    }
    return 0;
}

File: Beecrowd/2-AD-HOC/1250-KiloMan-2/1250.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int repeat = sc.nextInt();

        for (int r = 0; r < repeat; r++) {
            int ans = 0;
            int size = sc.nextInt();
            int[] disparos = new int[size];
            
            for (int i = 0; i < size; i++) {
                disparos[i] = sc.nextInt();
            }
            
            String jumps = sc.next();
            
            for (int i = 0; i < size; i++) {
                if ((jumps.charAt(i) == 'S' && disparos[i] <= 2) || 
                    (jumps.charAt(i) == 'J' && disparos[i] > 2)) {
                    ans++;
                }
            }
            System.out.println(ans);
        }
        sc.close();
    }
}
